### PR TITLE
ColorPickerDialog - migrate Dialog to use new implementation

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
ColorPickerDialog - migrate Dialog to use the new incubator implementation

## Changelog
ColorPickerDialog - migrate Dialog to use the new incubator implementation

### CODE REVIEW

File: src/components/colorPicker/ColorPickerDialog.tsx [hunk 1]
```diff
@@ -338,9 +338,9 @@ class ColorPickerDialog extends PureComponent<Props, State> {
         centerH
         onDismiss={this.onDismiss}
         containerStyle={styles.dialog}
-        panDirection={PanningProvider.Directions.DOWN}
+        direction={PanningProvider.Directions.DOWN}
```
Comment (M-i-k-e-l) [inline src/components/colorPicker/ColorPickerDialog.tsx:341]: This is not needed anymore

File: src/components/colorPicker/ColorPickerDialog.tsx [hunk 2]
```diff
@@ -338,9 +338,9 @@ class ColorPickerDialog extends PureComponent<Props, State> {
...
         testID={`${testID}.dialog`}
-        supportedOrientations={['portrait', 'landscape', 'landscape-left', 'landscape-right']} // iOS only
+        modalProps={{supportedOrientations: ['portrait', 'landscape', 'landscape-left', 'landscape-right']}} // iOS only
```
Comment (M-i-k-e-l) [inline src/components/colorPicker/ColorPickerDialog.tsx:343]: Can we make this a `const` please?

**Review by M-i-k-e-l — COMMENTED** [review]
When pressing done (the checkmark) the dialog is closed immediately without animation

File: src/components/colorPicker/ColorPickerDialog.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/components/colorPicker/ColorPickerDialog.tsx:343]: Not really, there's a type issue here since this is not a string array per se

Comment (M-i-k-e-l) [reply src/components/colorPicker/ColorPickerDialog.tsx:341]: I think it's best to add it as a const and cast it:
```
import {ModalProps} from '../../components/modal';
...
const MODAL_PROPS = {supportedOrientations: ['portrait', 'landscape', 'landscape-left', 'landscape-right']} as ModalProps;
```

**Review by M-i-k-e-l — APPROVED** [review]
Approved but I think we should make it a const (see the comment)

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**